### PR TITLE
Pattern Analysis: Analysis of ISIC dataset with Improved UNet 

### PR DESCRIPTION
**Pull request by james-seymour-cubiko**
Analyses the ISIC dataset with an improved UNet model. The model is split into 5 separate files:

- **dataset.py**: loads the ISIC dataset into training, validation and test `tf.data.Dataset` objects which can be passed to the model
- **utils.py**: defines a calculation for the DSC metric and the corresponding DSC loss
- **modules.py**: defines methods to produce the`tf.keras.layers.Layer` blocks / layers used in the model, with a UNet function to compose these blocks into the final model architecture
- **train.py**: trains the resulting model, and plots DSC history
- **predict.py**: uses the trained model to predict segmentation results from the test dataset and calculates the resulting DSC - also visualises prediction vs ground truth outputs

### CODE REVIEW

**Comment by SiyuLiu0329** [comment]
**This is an initial inspection, no action is required at this point**
- Code: OK
- Results: OK
- Readme: please also report quantitative results in readme; broken link on GitHub web 
- Commit Messages: OK
- Other comments: repo could use some clean up. eg put images in a folder.

**Comment by shakes76** [comment]
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate use and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem
Driver Script present
File structure present -1 (could be cleaner)
Shows Usage & Demo & Visualisation & Data usage -2 (no plots etc.)
Module present
Commenting
No Data leakage
Difficulty: Easy -10
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable/good
Good Description and Comments (sometimes minimal) -1
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good
